services/patient-greeting.js

```diff
@@ -0,0 +1,37 @@
+const express = require('express');
+
+const router = express.Router();
+
+// Middleware to verify all necessary data is available to process a response
+router.use((request, response, next) => {
+  const data = request.body;
+  if (!(data && data.prefetch && data.prefetch.patient && data.prefetch.patient.resource)) {
+    response.sendStatus(200);
+    return;
+  }
+  const patient = data.prefetch.patient.resource;
+  if (!(patient.name && patient.name[0] && patient.name[0].given && patient.name[0].given[0])) {
+    response.sendStatus(200);
+    return;
+  }
+  next();
+});
+
+// CDS Service endpoint
+router.post('/', (request, response) => {
+  const data = request.body;
+  const patient = data.prefetch.patient.resource;
+  const name = patient.name[0].given[0];
+  const card = {
+    cards: [{
+      summary: `Now seeing: ${name}`,
+      source: {
+        label: 'Patient greeting service',
+      },
+      indicator: 'info',
+    }],
+  };
+  response.send(JSON.stringify(card, null, 2));
```

Comment:
I think we should be using [response.json](http://expressjs.com/en/api.html#res.json) instead.

Reply:
the only difference is that what zach has is easier to read in browsers. It adds in the tabs like githubs apis.

Reply:
This line in index.js will accomplish this:

```javascript
app.set('json spaces', '  ')
```

Reply:
Ooh, nice 👍